src/openstudio_lib/LocationTabView.cpp

```diff
@@ -673,58 +929,112 @@ void LocationView::onDesignDayBtnClicked() {
+                if (dayType.contains("Winter")) {
+                  winterDays0_4.push_back(designDay);
+                } else if (dayType.contains("Summer")) {
+                  summerdays0_4.push_back(designDay);
+                }
+              } else {
+                unknownDay = true;
+              }
             } else {
-              unknownDay = true;
+              allNonAnnual.push_back(designDay);
             }
           }
         }
 
         // Pick only the most stringent design points
-        if (!unknownDay) {
-          if (!days99_6.empty()) {
-            for (model::DesignDay designDay : days99) {
-              designDay.remove();
-            }
-          }
-
-          if (!days0_4.empty()) {
-            for (model::DesignDay designDay : days1) {
-              designDay.remove();
-            }
-            for (model::DesignDay designDay : days2) {
-              designDay.remove();
-            }
-          } else if (!days1.empty()) {
-            for (model::DesignDay designDay : days2) {
-              designDay.remove();
-            }
-          }
-        }
+        // if (!unknownDay) {
+        //   if (!summerdays99_6.empty()) {
+        //     for (model::DesignDay designDay : summerdays99) {
+        //       designDay.remove();
+        //     }
+        //   }
+
+        //   if (!summerdays0_4.empty()) {
+        //     for (model::DesignDay designDay : summerdays1) {
+        //       designDay.remove();
+        //     }
+        //     for (model::DesignDay designDay : summerdays2) {
+        //       designDay.remove();
+        //     }
+        //   } else if (!summerdays1.empty()) {
+        //     for (model::DesignDay designDay : summerdays2) {
+        //       designDay.remove();
+        //     }
+        //   }
+
+        //   if (!winterDays99_6.empty()) {
+        //     for (model::DesignDay designDay : winterDays99) {
+        //       designDay.remove();
+        //     }
+        //   }
+
+        //   if (!winterDays0_4.empty()) {
+        //     for (model::DesignDay designDay : winterDays1) {
+        //       designDay.remove();
+        //     }
+        //     for (model::DesignDay designDay : winterDays2) {
+        //       designDay.remove();
+        //     }
+        //   } else if (!winterDays1.empty()) {
+        //     for (model::DesignDay designDay : winterDays2) {
+        //       designDay.remove();
+        //     }
+        //   }
+        // }
 
         // Evan note: do not remove existing design days
         //for (model::SizingPeriod sizingPeriod : m_model.getModelObjects<model::SizingPeriod>()){
         //  sizingPeriod.remove();
         //}
 
+        //m_model.insertObjects(ddyModel.objects());
+        designDaysToInsert = showDesignDaySelectionDialog(summerdays99, summerdays99_6, summerdays2, summerdays1, summerdays0_4, winterDays99, winterDays99_6, winterDays2, winterDays1, winterDays0_4,allNonAnnual);
```

Comment:
I would pass all the design days in in a single vector.  Then the widget could do all the sorting, doing all this pre-sorting up front is kind of messy.